tools/BuildAssets/BuildTasks/Microsoft.Azure.Sdk.Build.Tasks/SdkBuildTasks/ArchiveSymbolsTask.cs

```diff
@@ -0,0 +1,522 @@
+        /// False: Will create request file as well as execute to queue up symbol archive request
+        /// </summary>
+        public bool SkipExecuteSymbolsRequest { get; set; }
+        
+        /// <summary>
+        /// Project name registered with Symbol Archiving service
+        /// If not provided default name will be used set by the Azure .NET SDK team
+        /// </summary>
+        public string ArchiveSymbolsRequestProjectName { get; set; }
+
+        public ArchiveSymbolsTask()
+        {
+            SymbolsRequestFileList = new List<string>();
+            SkipExecuteSymbolsRequest = false;
+        }
+
+        public override bool Execute()
+        {
+            TaskLogger.LogInfo("Executing Archiving Task.....");
+
+            List<SymbolsCopyInfo> symbolList = GetSymbolsInfo();
+
+            foreach(SymbolsCopyInfo symInfo in symbolList)
+            {
+                symInfo.CreateSymbolRequest();
+                SymbolsRequestFileList.Add(symInfo.SymbolRequestFilePath);
+
+                if(SkipExecuteSymbolsRequest == false)
+                {
+                    symInfo.SendSymbolRequest();
+                    
+                    if(!string.IsNullOrWhiteSpace(symInfo.ArchiveRequestErrorOutput))
+                    {
+                        ApplicationException aEx = new ApplicationException(symInfo.ArchiveRequestErrorOutput);
+                        TaskLogger.LogException(aEx);
+                    }
+                    else
+                    {
+                        TaskLogger.LogInfo(symInfo.ArchiveRequestOutput);
+                    }
+                }
+            }
+
+            return true;
+        }
+
+        private List<SymbolsCopyInfo> GetSymbolsInfo()
+        {
+            // Symbol archive Path format: 12-12-2018\BuildJobId-4DigitRandomNumber\PackageName
+            int randomMaxValue = 9999;
+            Random rnd = new Random(DateTime.Now.Second);
+            string dateStr = DateTime.Now.ToString("dd-MM-yyyy");
+            string jobId = string.Format("{0}-{1}", BuildJobId, rnd.Next(randomMaxValue));
+            int requestIdCount = 1;
+
+            List<SymbolsCopyInfo> symReqList = new List<SymbolsCopyInfo>();
+            Dictionary<string, string> uniqueDirPath = new Dictionary<string, string>();
+            // Iterate all the file path
+            // First get the "release" directory each release directory has target framework directory
+            // for e.g. net452, netstandard1.4, these directories will be copied to the symbols directory, because the symbol service will have to
+            // archive copy both target symbols
+            // We then go two level up, that will give us the ResourceProvider directory
+            // This directory helps in creating distinct directory names to avoid collision in scenario where there are more than
+            // one nuget package is getting published
+
+            foreach(string asmFile in BuiltAssemblyList)
```

Comment:
Have you considered just taking the symbols out of the symbol packages instead?

Reply:
1.  think it is somewhat wasteful to blindly copy all these files.
2. I'm pretty sure the pdb2pdb tool doesn't require anything beyond the pdb (see https://github.com/dotnet/symreader-converter).

Reply:
That said I still need to dig in further to fully understand our symbol publishing process to ensure we are doing the right thing.